include/envoy/stats/stats.h

```diff
@@ -146,12 +150,27 @@ class Gauge : public virtual Metric {
   virtual void inc() PURE;
   virtual void set(uint64_t value) PURE;
   virtual void sub(uint64_t amount) PURE;
-  virtual bool used() const PURE;
   virtual uint64_t value() const PURE;
 };
 
 typedef std::shared_ptr<Gauge> GaugeSharedPtr;
 
+/**
+ * Holds the computed statistics for a histogram.
+ */
+class HistogramStatistics {
+public:
+  virtual ~HistogramStatistics() {}
+
+  /**
+   * Returns summary representation of the histogram.
+   */
+  virtual std::string summary() const PURE;
+
+  double quantiles_in_[9] = {0, 0.25, 0.5, 0.75, 0.90, 0.95, 0.99, 0.999, 1};
```

Comment:
These are an implementation detail right now, right? Can we move this out of the interface and into the implementation?

Reply:
i am using these in MetricsService otherwise I have to use dynamic_cast to impl and access.

Reply:
Maybe use a pure method?

  double* quantiles() const PURE;
  size_t numQuantiles() const PURE;

or you could return std::pair<double*, size_t> -- not sure which is preferred.

Reply:
+1, what about just returning a `const std::vector<double>&` via a pure virtual method? This can be hidden by the implementation.

Reply:
changed it to use a pure virtual method returning const std::vector<double>&